### PR TITLE
Task 5 (AWS Simple Storage Service - S3)

### PR DESCRIPTION
**Pull request by aleksandr-zharov**
https://github.com/rolling-scopes-school/nodejs-aws-tasks/blob/main/task5-import-to-s3/task.md

FE: https://dndbrxqx66t48.cloudfront.net/admin/products
API: https://z8fgw3vdoc.execute-api.eu-west-1.amazonaws.com/dev/import?name=test.csv

Scope:
 1 - File serverless.yml contains configuration for importProductsFile function
 3 - The importProductsFile lambda function returns a correct response which can be used to upload a file into the S3 bucket
 4 - Frontend application is integrated with importProductsFile lambda
 5 - The importFileParser lambda function is implemented and serverless.yml contains configuration for the lambda
Additional scope:
 +1 - async/await is used in lambda functions
 +1 - importProductsFile lambda is covered by unit tests (aws-sdk-mock can be used to mock S3 methods - https://www.npmjs.com/package/aws-sdk-mock)
image
 +1 - At the end of the stream the lambda function should move the file from the uploaded folder into the parsed folder

### CODE REVIEW

**Comment by kotmatpockuh** [comment]
please provide links later :)

**Comment by aleksandr-zharov** [comment]
> please provide links later :)

added

File: import-service/common/default.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
+export const successHandler = (result) => {
+	return {
+		statusCode: 200,
+		headers: {
+			'Content-Type': 'application/json',
+			'Access-Control-Allow-Origin': '*',
```
Comment (kotmatpockuh) [inline import-service/common/default.ts:6]: I guess you don't need all methods? 🤔

File: import-service/handlers/import-file-parser.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
+import { S3 } from 'aws-sdk';
+import {S3Event} from 'aws-lambda';
```
Comment (kotmatpockuh) [inline import-service/handlers/import-file-parser.ts:2]: try to keep consistent formatting ;)

File: import-service/handlers/import-products-file.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+import { errorHandler, successHandler } from '../common/default';
+import {DefaultError} from "../common/errors";
+
+const bucketName = process.env.S3_BUCKET_NAME;
+const region = process.env.S3_REGION;
+
+
+export const handler : APIGatewayProxyHandler = async (event) => {
+  try {
+    if (!event.queryStringParameters || !event.queryStringParameters.name) {
```
Comment (kotmatpockuh) [inline import-service/handlers/import-products-file.ts:12]: You are using TS. So you can write something like: `!event?.queryStringParameters?.name`

File: import-service/handlers/import-file-parser.ts (hunk 2 above)
Comment (aleksandr-zharov) [reply import-service/handlers/import-file-parser.ts:2]: Thanks for comments. I'll be more accurate.

**Pull request merged**